qiskit/providers/fake_provider/fake_generic.py

```diff
@@ -0,0 +1,612 @@
+                f"The number of qubits (got {num_qubits}) must match "
+                f"the size of the provided coupling map (got {coupling_map.size()})."
+            )
+
+        self._num_qubits = num_qubits
+        self._coupling_map = coupling_map
+
+        # Hardcoded default target attributes. To modify,
+        # access corresponding properties through the public `Target` API
+        super().__init__(
+            description=f"Generic Target with {num_qubits} qubits",
+            num_qubits=num_qubits,
+            dt=0.222e-9,
+            qubit_properties=[
+                QubitProperties(
+                    t1=self._rng.uniform(100e-6, 200e-6),
+                    t2=self._rng.uniform(100e-6, 200e-6),
+                    frequency=self._rng.uniform(5e9, 5.5e9),
+                )
+                for _ in range(num_qubits)
+            ],
+            concurrent_measurements=[list(range(num_qubits))],
+        )
+
+        # Ensure that Reset, Delay and Measure are in
+        # self._basis_gates
+        # so that their instructions are added to the target.
+        self._basis_gates = basis_gates
+        for name in ["reset", "delay", "measure"]:
+            if name not in self._basis_gates:
+                self._basis_gates.append(name)
+
+        # Iterate over gates, generate noise params from defaults,
+        # and add instructions to target.
+        for name in self._basis_gates:
+            if name not in self.supported_operations:
+                raise QiskitError(
+                    f"Provided base gate {name} is not a supported "
+                    f"operation ({self.supported_operations.keys()})."
+                )
+            gate = self.supported_operations[name]
+            noise_params = self.noise_defaults[name]
+            self.add_noisy_instruction(gate, noise_params)
+
+        if control_flow:
+            self.add_instruction(IfElseOp, name="if_else")
+            self.add_instruction(WhileLoopOp, name="while_loop")
+            self.add_instruction(ForLoopOp, name="for_loop")
+            self.add_instruction(SwitchCaseOp, name="switch_case")
+            self.add_instruction(BreakLoopOp, name="break")
+            self.add_instruction(ContinueLoopOp, name="continue")
+
+        # Generate block of calibration defaults and add to target.
+        # Note: this could be improved if we could generate and add
+        # calibration defaults per-gate, and not as a block.
+        defaults = self._generate_calibration_defaults(calibrate_instructions)
+        inst_map = defaults.instruction_schedule_map
+        self.add_calibrations_from_instruction_schedule_map(inst_map)
+
+    @property
+    def supported_operations(self) -> dict[str, Instruction]:
+        """Mapping of names to class instances for operations supported
+        in ``basis_gates``.
+
+        Returns:
+            Dictionary mapping operation names to class instances.
+        """
+        return {
```

Comment:
Partially fixed in [982deb7](https://github.com/Qiskit/qiskit/pull/10266/commits/982deb763795f9d2e0e8af26868fb2105318f32b), I kept the noise defaults as a public property because I thought that users might want to query it to know what noise values have been used.